source/codegen/templates/proto_helpers.mako

```diff
@@ -53,6 +53,48 @@ enum ${enum_name} {
 % endfor
 </%def>
 
+## Define enums in the proto per attribute type that uses enum values
+<%def name="define_attribute_values_enums(attribute_enums)">\
+<%
+  config = data["config"]
+  service_class_prefix = config["service_class_prefix"]
+  enums = data["enums"]
+%>\
+% for type_name in attribute_enums:
+<%
+  type_enum_name = service_class_prefix + type_name[2:] + "AttributeValues"
+  enum_value_type_specific_prefix = (f"{service_class_prefix.upper()}_{type_name[2:]}_VAL").upper()
+  values_to_create = { enum_value_type_specific_prefix + "_UNSPECIFIED": 0 }
+  for enum_name in sorted(attribute_enums[type_name]):
+    enum_value_prefix = enum_value_type_specific_prefix + "_" + common_helpers.pascal_to_snake(enum_name)
+    enum = enums[enum_name]
+    nonint_index = 1
+    for value in enum["values"]:
+      enum_value_suffix = value['name'].replace(f"{service_class_prefix.upper()}_VAL_", "")
+      value_name = f"{enum_value_prefix}_{enum_value_suffix}".upper()
+      if enum.get("generate-mappings", False):
```

Comment:
If one of the values needs to generate a mapping, then all of them need to be mapped. You can't do mapping on a case by case basis.

Reply:
Yes, I have updated code to do `generate-mappings` by looking for duplicate values over entire consolidated enum, instead of looking at individual split enums.